Source/WebCore/PAL/pal/text/TextCodecCJK.cpp

```diff
@@ -899,7 +899,7 @@ static const GB18030EncodeIndex& gb18030EncodeIndex()
 // https://unicode-org.atlassian.net/browse/ICU-22357
 // The 2-byte values are handled correctly by values from gb18030()
 // but these need to be exceptions from gb18030Ranges().
-static std::optional<uint32_t> gb18030AsymmetricEncode(char32_t codePoint)
+static std::optional<uint16_t> gb18030AsymmetricEncode(UChar codePoint)
```

Comment:
Please carefully review these changes. As far as I can tell it's never called with anything but a UChar, so not sure why we'd take a char32_t. And it only needs to return two bytes.